pkg/input/remoteread/remoteread.go

```diff
@@ -1,3 +1,242 @@
 package remoteread
 
-// TBD, help wanted!
+import (
+	"context"
+	"github.com/go-kit/kit/log"
+	"github.com/pkg/errors"
+	config_util "github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/pkg/timestamp"
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/prometheus/prometheus/storage/remote"
+	"github.com/prometheus/prometheus/tsdb/chunkenc"
+	"github.com/thanos-community/obslytics/pkg/input"
+	"github.com/thanos-community/obslytics/pkg/version"
+	"net/url"
+
+	"time"
+)
+
+type remoteReadInput struct {
```

Comment:
Since this package is called remote read this might be called just  
"Input"

 (: 

Also worth to rename remote read package with "promread" so it indicates we Prometheus API (:

Reply:
I was just trying to copy the storeAPI format, https://github.com/thanos-community/obslytics/blob/master/pkg/input/storeapi/storeapi.go#L24